src/main/java/com/worldline/codetest/FatJarApplication.java

```diff
@@ -17,23 +22,26 @@
  */
 public class FatJarApplication extends Application {
 
-    HashSet<Object> singletons = new HashSet<Object>();
+    private static final HashSet<Object> singletons = new HashSet<Object>();
+    private static final HashSet<Class<?>> classes = new HashSet<Class<?>>();
 
     public FatJarApplication() {
+        ProfileService profileService = new ProfileService(new ProfileRepositoryMapImp());
```

Comment:
Nice Refactoring, I wasn't aware about the `getSingeltion`() method could be used to create Singelton classes

Reply:
What we can conclude here is that in JAX-RS

- Singletons are already constructed API:s

- Classes are hint to JAX-RS specification implementation ( RestEasy or other implementations) on what to be initiated with a default empty constructor ( requirement ).

take away lesson here is that dependency injection is for free in Java, aka Constructors.